code/game/mecha/equipment/mecha_equipment.dm

```diff
@@ -48,7 +48,10 @@
 	return ..()
 
 /obj/item/mecha_parts/mecha_equipment/try_attach_part(mob/user, obj/mecha/M)
-	if(can_attach(M))
+	if(!can_attach(M))
+		return FALSE
+	 if(!do_mob(user, M, 15))
```

Comment:
bruh is that a single space

Reply:
The single space was added with the suggestions, I believe, currently fixing